### DISCUSSION TITLE
Does dynamic block are supported ?

### DISCUSSION THREAD

**Discussion by pferandin** [DomCR/ACadSharp#406] [Q&A]
Hi, 

I have a DWG files with dynamic blocks insided, when i copy it using DWGReader and write it using DWGWriter the output files cannot be openned. 
Loading is stuck at 99% when openning it using ZWCAD.

**Comment by DomCR** [comment]
Hi @pferandin,

Dynamic blocks are not yet supported in `ACadSharp`, the information I think is in the `XRecords`, but there are no tools right no to work with them.

I'm not familiar with ZWCAD, is this issue being caused by a particular file generated by `ACadSharp` or it happens with all of them?

**Reply by pferandin** [reply]
Re @DomCR, 

Yes the issue is caused by a particular file, but not really an AcadSharp problem, i think i had the problem because the plan contains many dynamic block. 

I have tried with another dwg file, without any dynamic block and there is no problem to open it using ZWCAD after a clone operation using AcadSharp. 

Do you think it'll be possible to work with dynamic block ? 
If yes, is this feature planned in your roadmap ? 
Have you any documentation about this ? I don't find any documentation about DynamicBlock in DXF documentation 😢

Maybe i can contribute to supporting dynamic block in AcadSharp, i'll investigate 👌

Regards,

**Reply by DomCR** [reply]
> Do you think it'll be possible to work with dynamic block ?

I think dxf allows to save dynamic blocks, if that's the case it would be possible to work with them, for dwg will be more complicated, I could not find any available documentation about it so it will be tricky. 

> If yes, is this feature planned in your roadmap ?

Right now I'm focused on stabilizing the project and adding the last main features to release the v1, but in the future I'll like to add the feature for sure.

Have you any documentation about this ? I don't find any documentation about DynamicBlock in DXF documentation 😢

I think that if you want to explore this issue you will have to "manually" read a dxf file and see what's in it, I've found this answer that explains how dxf saves the dynamic blocks.

https://forums.autodesk.com/t5/autocad-forum/dynamic-blocks-in-dxf/m-p/9567892#M1022687

I'll open a branch attached to this discussion to keep in mind the investigation.

**Reply by mme1950** [reply]
see [issue at nanoLogika/ACadSharp](https://github.com/nanoLogika/ACadSharp/issues/6)
